contracts/adapters/IPRebalanceExtension.sol

```diff
@@ -0,0 +1,478 @@
+        uint256 minTransformDelay;          // minumum delay between transfroms/untransforms
+        ITransformHelper transformHelper;   // TransformHelper contract address
+    }
+
+    /* ========== State Variables ========= */
+
+    IAirdropModule public airdropModule;
+
+    // mapping from transform component to TransformInfo
+    mapping(address => TransformInfo) public transformComponentInfo;
+    // mapping from set component to target units
+    mapping(address => uint256) public rebalanceParams;
+
+    // mapping from transform component to last transform/untransform timestamp
+    mapping(address => uint256) public lastTransform;
+
+    // list of all set components involved in rebalance including added/removed components
+    address[] public setComponentList;
+
+    // flag marking whether GIM trades have completed
+    bool public tradesComplete;
+
+    /* ========== Constructor ========== */
+
+    /**
+     * Sets requires state variables
+     *
+     * @param _manager              BaseManagerV2 manager contract
+     * @param _generalIndexModule   Set Protocol GeneralIndexModule
+     * @param _airdropModule        Set Protocol AirdropModule
+     */
+    constructor(
+        IBaseManager _manager,
+        IGeneralIndexModule _generalIndexModule,
+        IAirdropModule _airdropModule
+    )
+        public
+        GIMExtension(_manager, _generalIndexModule)
+    {
+        airdropModule = _airdropModule;
+    }
+
+    /* ======== External Functions ======== */
+
+    /**
+     * Original function for starting rebalances from the inherited GIMExtension. This function has been deprecated
+     * in favor of startIPRebalance. It will revert when it is called.
+     */
+    function startRebalanceWithUnits(
+        address[] calldata /* _components */,
+        uint256[] calldata /* _targetUnitsUnderlying */,
+        uint256 /* _posotionMultiplier */
+    )
+        external
+        onlyOperator
+        override
+    {
+        revert("use startIPRebalance instead");
+    }
+
+    /**
+     * ONLY OPERATOR: Sets a TransformInfo entry for a transform component. Each transform component must have a TransformInfo set
+     * for it which includes neccessary information such as the underlying component and TransformHelper address. Must not already
+     * have been set. If it has, use updateTransformInfo.
+     *
+     * @param _transformComponent       component that needs to be transformed/untransformed during rebalance
+     * @param _transformInfo            TransfomInfo entry for the transform component
+     */
+    function setTransformInfo(address _transformComponent, TransformInfo memory _transformInfo) external onlyOperator {
+        require(
+            transformComponentInfo[_transformComponent].underlyingComponent == address(0),
+            "TransformInfo already set"
+        );
+        
+        require(_transformInfo.maxTransformSize <= type(uint96).max, "max transform size must be less than MAX_UINT_96");
```

Comment:
Why use `uint96` here? Would add a comment explaining since that's a break from our usual typing.